LibGit2Sharp/Branch.cs

```diff
@@ -226,7 +236,17 @@ private Branch ResolveTrackedBranch()
                 return null;
             }
 
-            string trackedReferenceName = Proxy.git_branch_upstream_name(repo.Handle, CanonicalName);
+            string trackedReferenceName = null;
+            try
+            {
+                trackedReferenceName = Proxy.git_branch_upstream_name(repo.Handle, CanonicalName);
+            }
+            catch (InvalidSpecificationException)
+            {
+                // we have no way to pro-actively check this
```

Comment:
So there's two scenarios here which can trigger this exception:

 - `CanonicalName` is an invalid refspec (that's my dumb error)
 - the nick configured under `branch.master.remote` is a full URL, which is parsed incorrectly in libgit2

I'd rather `libgit2` be responsible for this (it should return null for this scenario as upstream doesn't exist) but I don't want to hold up this release with patching libgit2 for this obscure shit. Also, I haven't verified this against the latest release of libgit2.

> But won't this mean that we return null even though we're tracking a branch?

No tracking information is kept for this branch when you set it to the full URL. It's hella dumb to set it up this way - but you can.